pyproject.toml

```diff
@@ -0,0 +1,248 @@
+maintainers = [{name = "Matthew Rocklin", email = "mrocklin@gmail.com"}]
+license = {text = "BSD"}
+classifiers = [
+    "Development Status :: 5 - Production/Stable",
+    "Intended Audience :: Developers",
+    "Intended Audience :: Science/Research",
+    "License :: OSI Approved :: BSD License",
+    "Operating System :: OS Independent",
+    "Programming Language :: Python",
+    "Programming Language :: Python :: 3",
+    "Programming Language :: Python :: 3 :: Only",
+    "Programming Language :: Python :: 3.8",
+    "Programming Language :: Python :: 3.9",
+    "Programming Language :: Python :: 3.10",
+    "Programming Language :: Python :: 3.11",
+    "Topic :: Scientific/Engineering",
+    "Topic :: System :: Distributed Computing",
+]
+readme = "README.rst"
+requires-python = ">=3.8"
+dependencies = [
+    "click >= 8.0",
+    "cloudpickle >= 1.5.0",
+    "dask == 2023.3.2",
+    "jinja2 >= 2.10.3",
+    "locket >= 1.0.0",
+    "msgpack >= 1.0.0",
+    "packaging >= 20.0",
+    "psutil >= 5.7.0",
+    "pyyaml >= 5.3.1",
+    "sortedcontainers >= 2.0.5",
+    "tblib >= 1.6.0",
+    "toolz >= 0.10.0",
+    "tornado >= 6.0.3",
+    "urllib3 >= 1.24.3",
+    "zict >= 2.2.0",
+]
+dynamic = ["version"]
+
+[project.urls]
+Homepage = "https://distributed.dask.org"
+Source = "https://github.com/dask/distributed"
+
+[project.entry-points.dask_cli]
+scheduler = "distributed.cli.dask_scheduler:main"
+worker = "distributed.cli.dask_worker:main"
+ssh = "distributed.cli.dask_ssh:main"
+spec = "distributed.cli.dask_spec:main"
+
+[project.scripts]
+dask-ssh = "distributed.cli.dask_ssh:main"
+dask-scheduler = "distributed.cli.dask_scheduler:main"
+dask-worker = "distributed.cli.dask_worker:main"
+
+[tool.setuptools]
+include-package-data = true
+zip-safe = false
+
+[tool.setuptools.packages.find]
+exclude = ["*tests*"]
+namespaces = false
+
+[tool.setuptools.package-data]
+"*" = [
+    "templates/index.html",
+    "template.html",
+]
+distributed = [
+    "http/templates/*.html",
+    "py.typed",
+]
+
+[tool.isort]
+sections = ["FUTURE", "STDLIB", "THIRDPARTY", "DISTRIBUTED", "FIRSTPARTY", "LOCALFOLDER"]
+profile = "black"
+skip_gitignore = true
+force_to_top = ["true"]
+default_section = "THIRDPARTY"
+known_first_party = ["distributed"]
+known_distributed = ["dask", "zict"]
+add_imports = ["from __future__ import annotations"]
+
+[tool.versioneer]
+VCS = "git"
+style = "pep440"
+versionfile_source = "distributed/_version.py"
+versionfile_build = "distributed/_version.py"
+tag_prefix = ""
+parentdir_prefix = "distributed-"
+
+[tool.pytest.ini_options]
+addopts = '''
+-v -rsxfE
+--durations=20
+--color=yes
+--ignore=continuous_integration
+--ignore=docs
+--ignore=.github
+--strict-markers
+--strict-config
+-p no:asyncio
+-p no:legacypath'''
+filterwarnings = [
+    "error",
+    '''ignore:Please use `dok_matrix` from the `scipy\.sparse` namespace, the `scipy\.sparse\.dok` namespace is deprecated.:DeprecationWarning''',
```

Comment:
Nit for readability - is there a reason all of these are triple-single-quoted? Assuming some of these are for character escaping purposes (not too familiar with TOML syntax), but this example pyproject.toml seems to imply that a lot of these could be double quoted instead:

https://github.com/rapidsai/dask-cuda/blob/9fef6b7b1a65ab0a5bd5784c440451d8fcb2d71e/pyproject.toml#L121-L131

Reply:
A few of these have literal backslash escapes and single quotes in and I wanted the indentation to line up across all of them

Reply:
Ah thanks for the clarification, didn't realize that pytest requires literal backslashes in the filter strings. In that case understand that it's probably best to stick to one TOML string type and stay consistent